### PR TITLE
Added HOC for data-Seleniumid

### LINKED ISSUE #641: Add data attribute to component for selenium
- [ ] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

## Are you reporting a bug or a feature?

 - [ ] Bug
 - [x] Feature

Axa has a selenium toolkit project https://github.axa.com/eClaims-SV/selenium-toolkit 
It works if components have an unique `data-seleniumid` ~~tag~~ attribute on them for read/ write access. On the project schadenmeldung claims we used it already with an old branch by mirroring the id (see [dist-3](https://github.com/axa-ch/patterns-library/blob/form-components-with-dist3/src/components/a-form-input/_template.js
)). This should not break any existing functionality that way and even improve slightly the code by forcing the coder to supply an id to each used component. (If none is set, the selenium tests won't work since the random ui id can't be used as identifier to get the component)

The toolkit creates a testreport, creates java classes for easy access to the components and provides some generic methods for setting/ getting the data, handles the browsers and co.
There is a small playground/ demo project available [demo](https://github.axa.com/eClaims-SV/selenium-playground)

### PR DESCRIPTION
**Pull request by michel-ho**
Fixes #641.

Changes proposed in this pull request:

 - Adds new HOC (if you use this, data-seleniumid will be setted by id)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
 
 # How Has This Been Tested?
 
 
 
 # Checklist:
 
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my own code
 - [ ] I have commented my code, particularly in hard-to-understand areas
 - [ ] I have made corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/js/abstract/hocs/with-data-seleniumid.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+   * This is the recommended approach.
+   */
+  class WithDataSeleniumId extends Base {
+
+    connectedCallback() {
+      if (super.connectedCallback) {
+        super.connectedCallback();
+      }
+
+      const { constructor: { tagName } } = this;
```
Comment (AndyOGo) [inline src/js/abstract/hocs/with-data-seleniumid.js]: Great.
Now we can discuss if we really still need additional checks if this feature shall be enabled.
IMHO not necessarily.

More important is that we document how Selenium selects `<custom-elements>` and how it interacts with its children.

@LucaMele 
I see one challenge here, ideally it's possible to enable it from outside of any component, for example by:
- `window.__enableDataSeleniumId` ???
- one more HTML attribute for each component, like `selenium-id-enabled`

File: src/js/abstract/hocs/with-data-seleniumid.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const withDataSeleniumId = Base =>
+  /**
+   * Base class {WithBaseGlobal}. This class extends the {Base} and
```
Comment (AndyOGo) [inline src/js/abstract/hocs/with-data-seleniumid.js]: doclet needs to be updated

File: src/js/abstract/hocs/with-data-seleniumid.js (hunk 1 above)
Comment (LucaMele) [reply src/js/abstract/hocs/with-data-seleniumid.js]: @AndyOGo i do like the idea of having `window.__enableDataSeleniumId` setting.

Comment (AndyOGo) [reply src/js/abstract/hocs/with-data-seleniumid.js]: Alright.

So @michel-ho let's go with that

**Pull request closed without merging**